src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -468,7 +468,7 @@ public static LicenseStatus GetLicenseStatus(License license)
             return licenseStatus;
         }
 
-        public void TryActivateLicense(bool throwOnActivationFailure)
+        public async ValueTask TryActivateLicenseAsync(bool throwOnActivationFailure)
```

Comment:
Why is this an `async ValueTask`?

Reply:
`TryActivateLicenseAsync` just like `EnsureNotPassiveAsync` are expected to return without actually any `async` work.

Reply:
See here: https://blog.marcgravell.com/2019/08/prefer-valuetask-to-task-always-and.html

In general, if we have to call `await` internally, that isn't saving a lot.
And if we are going to await internally and outside, there is little point in here.
Also, this isn't likely to be in anything that is in the hot path

Reply:
I read this and others few times. TBH I'm still confused :-)

As I understand, the bottom line of that post is to use `ValueTask` _always_ (it is even in the title). But to be aware of the API changes.

Yet, as long as `AsyncTaskMethodBuilder` and friends aren't showing up in the profiler `Task` should be fine to use.

Reply:
In general, in our internal API, prefer `Task` unless we know that:

* This is called frequently.
* We expect (and will write code) to handle the completion in a synchronous manner.

For external API:

* If they are likely to complete with no async work, use `ValueTask`